apps/speed.c

```diff
@@ -1778,7 +1778,13 @@ int speed_main(int argc, char **argv)
         buflen = lengths[size_num - 1];
         if (buflen < 36)    /* size of random vector in RSA benchmark */
             buflen = 36;
-        buflen += MAX_MISALIGNMENT + 1;
+        if (INT_MAX - (MAX_MISALIGNMENT + 1) < buflen)
+        {
+            BIO_printf(bio_err, "Error: buffer size too large\n");
+            goto end;
+        } else {
```

Comment:
This could actually be not an "else" at all and just unconditional since the main branch ends in a `goto` so will never get here anyway.